### ISSUE TITLE
Error building class name

### ISSUE THREAD

**Issue by nuno-andre** [koxudaxi/datamodel-code-generator#243]
Parsing [this schema](https://csrc.nist.gov/schema/cpe/feed/1.0/nvd_cpe_feed_json_1.0.schema): 

```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "JSON Schema for NVD Common Product Enumeration (CPE) Feed version 0.1 Beta",
    "id": "https://scap.nist.gov/schema/nvd/feed/1.0/nvd_cve_feed_json_1.0_beta.schema",
    "description": "NVD data feed of CPE",
    ...
}
```

raises `black.InvalidInput: Cannot parse: 10:6: class 1 Beta(BaseModel):`

The problem was settled by changing the schema's title to `Cpe`.

**Version:**
 - OS: Windows
 - Python version: `3.7.7`
 - datamodel-code-generator version: `0.6.0`

**Comment by koxudaxi** [comment]
Thank you for creating this issue.
This code-generator uses the title of JSONSchema as the root model name.
But, It doesn't care about a special characters like space.
I will fix it.

**Comment by nuno-andre** [comment]
Thanks to you for this package. I can't think of a better output for an openapi/jsonschema codegen for Python than Pydantic.

As an idea, would be `id` (if there is one) a better candidate than `tittle`? Its restrictions are much more like the Python identifiers rules and tends to be more synthetic. In this case, for example, removing version numbers (`1.0_beta') and superflous words (`json`) from filename, and converting to PascalCase, would look like `NvdCveFeed`. Joining the path elements with similar criteria could be a another option.

**Comment by koxudaxi** [comment]
@nuno-andre 
Your idea is good. `id` is more usable for the model name. 

I thought removing disallow characters from the name.
But, the model name must be a unique name. Also, Users need to able to find the model by name.
we should impliment the naming rule. :thinking: 
https://github.com/koxudaxi/datamodel-code-generator/blob/b4a815702bcbd94977d86a6099d19280c3a113a0/datamodel_code_generator/reference.py#L142-L149

**Comment by nuno-andre** [comment]
The main sticking point is that `id`, `title`, or `description` are not necessary for a JSON schema to be valid; so it's possible to reach a dead end. I think a good approach could be:

1. Enabling an optional user input to hard coding the name, e.g. `--classname`.
2. Raise an explicit error if `get_valid_name` fails that asks the user to add this parameter.
3. Progressively refining an idempotent algorithm to extract the name from the available sources ordered by their probable usefulness.  Maybe `id`, file name, `title`, `description` ?

Also, it could be a good idea to normalize the string first to avoid losing information in the process, especially with `title` or `description`. A very rough example:

```python3
from unicodedata import normalize, combining
from itertools import filterfalse
import re

def get_valid_name(self, name: str, camel: bool = False) -> str:
    # normal form
    output = normalize('NFKD', name)
    output = ''.join(filterfalse(combining, output))

    # tokenization before substitution to keep the structure
    output = re.split(r'[\s_-]', output)   # surely more will be needed

    # clean tokens - a symbol instead of '' would allow a later refination
    output = (re.sub('[^a-z0-9]', '', t.casefold()) for t in output)

    # purge useless tokens
    output = filter(is_useful, output)

    if camel:
        return ''.join(map(str.capitalize, output))

    return '_'.join(parts)
```

`is_useful` could range from something like

```python3
USELESS_WORDS = {'json', 'schema', 'api', 'version', 'for', 'of', 'the'}

def is_useful(token: str) -> bool:
    return token not in USELESS_WORDS
```

to a more sophisticated function for the entire list to evaluate the token in context.

**Comment by koxudaxi** [comment]
@nuno-andre
Thank you for your good approach and the example.
OK, I will implement it.

**Comment by koxudaxi** [comment]
@nuno-andre 
I have released a new  version as `0.6.4`

This release has two changes.
1. add `--class-name` option sets the name of the root model.
2. raise an error when the model name(title) is invalid and `--class-name` is not set.

I considered your approach that is below.
1. use `id` and `description` as a model name.
2. improve `get_valid_name`

I feel good about the approach. But, this is a breaking change. 
I don't choose the breaking change in this release. 

However, I want to implement these next major(1.x) or miner (0.7.x) version. 

Thank you

**Comment by nuno-andre** [comment]
Great. Thank you!

Moving forward to avoid dead spots can be a great improvement to generate models also dynamically. I'll stay tuned.
